examples/compare_solvers.jl

```diff
@@ -8,6 +8,7 @@ using AtomsBuilder
 using DFTK
 using LinearAlgebra
 using PseudoPotentialData
+using Manopt, Manifolds, RecursiveArrayTools
```

Comment:
It's a little weird that one needs to `using` all there three packages explicitly. Can we somehow avoid that ?

Reply:
Uff. I would have no clue how to do that?
I also do not agree that that is inconvenient, it is what the extensions now bring – more modular packages without direct dependencies:

One can use Manopt with an own manifold (nand then not load Manifolds)
One can use Manifolds with an own implementation for the product structure (and then not load RecursiveArrayTools)

What would that package look like? Something with the Reexport all package? I am not sure what to do here.